### DISCUSSION TITLE
Gradient Accumulation: dealing with LR and Effective Batch Size

### DISCUSSION THREAD

**Discussion by sh3rlock14** [Lightning-AI/pytorch-lightning#17434] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule]
Hi everybody 👋🏻 
I've some doubts when dealing with Gradient Accumulation in a single GPU scenario:

Let's say that I use `Trainer(accumulate_grad_batches=k)`, and let `b` be the batch_size of my `DataLoader`.

1. I've read [this discussion](https://github.com/Lightning-AI/lightning/discussions/17035), but to me it is still unclear if an updated version of the LR (accordingly to `k`) is used and whether the `optimizer` handles it automatically or not.
2. If I'am using `LightningModule.log(..., batch_size=?)`, what is the Effective Batch Size to be passed as `batch_size`? Is it `b*k` or simply `b`?

Thanks for your help!
